src/stats/logger.go

```diff
@@ -0,0 +1,28 @@
+package stats
+
+import (
+	"strconv"
+	"time"
+
+	logger "github.com/sirupsen/logrus"
+)
+
+const logTemplate = `{"level":"debug","ts":%[1]v,"logger":"gostats.loggingsink","msg":"flushing %[2]v","json":{"name":"%[3]v","type":"%[2]v","value":"%[4]v"}}`
+
+// LoggingSink is a gostats.FlushableSink that logs all stats using
+// the rate limit logger.
+type LoggingSink struct{}
+
+func (s *LoggingSink) log(name, typ string, value float64) {
+	if logger.IsLevelEnabled(logger.DebugLevel) {
+		sec := float64(time.Now().UnixNano()) / float64(time.Second)
+		logger.Debugf(logTemplate, strconv.FormatFloat(sec, 'f', 6, 64), typ, name, value)
+	}
+}
+
+// Implementation of the gostats.FlushableSink interface
+
+func (s *LoggingSink) FlushCounter(name string, value uint64) { s.log(name, "counter", float64(value)) }
+func (s *LoggingSink) FlushGauge(name string, value uint64)   { s.log(name, "gauge", float64(value)) }
+func (s *LoggingSink) FlushTimer(name string, value float64)  { s.log(name, "timer", value) }
+func (s *LoggingSink) Flush()                                 { s.log("", "all stats", 0) }
```

Comment:
Sorry what does this part do? Are we going to end up logging counters and stuff?

Reply:
This is actually the current behavior when users start the ratelimit service with `DisableStats = false && UseStatsd = false`. It initializes a default gostats logging sink that prints all those logs to the standard output.

There are several minor issues asking for a way to silence those logs when that config combo is used, and this PR attempts to provide that by bridging the gostats to the ratelimit logger instead of writing them directly to the stdout, to allow users leverage the LOG_LEVEL setting accordingly.

One could argue that if users want to silence those logs, they should just set `DisableStats = true`, and just close this PR, but I think it is nice to have these gosats logs honor the ratelimit logging config anyway. I don't have a strong opinion, though, so happy to close the PR if this is something not worth it :)

Reply:
I would just nop these functions personally since I can't see how these logs have any value whatsoever. The rest seems fine.